dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/metrics/ChartCreator.java

```diff
@@ -6,88 +6,93 @@
-    }
-
-    public BarChartModel changeToMonthlyModel(int selectedYear) {
-        List<DatasetsMetrics> monthlyDatasetStats =
-                MetricsUtil.fillMissingDatasetMonths(metricsServiceBean.countPublishedDatasets(), selectedYear);
+        String xLabel = BundleUtil.getStringFromBundle("metrics.year");
+        String yLabel = BundleUtil.getStringFromBundle("metrics.chart.legend.label." + chartType);
+        String title = BundleUtil.getStringFromBundle("metrics.chart.title." + chartType);
 
-        return createBarModel(monthlyDatasetStats, BundleUtil.getStringFromBundle("metrics.month")
-                , initMonthlyBarModel(monthlyDatasetStats));
+        return createBarModel(yearlyMetrics, title, xLabel, yLabel, chartMode);
     }
 
-    BarChartModel initYearlyBarModel(List<DatasetsMetrics> datasets) {
-        BarChartModel model = new BarChartModel();
+    public BarChartModel createMonthlyChart(List<ChartMetrics> metrics, int year, String chartType, String chartMode) {
+        List<ChartMetrics> monthlyChartStats =
+                MetricsUtil.fillMissingMonthsForMetrics(metrics, year);
 
-        ChartSeries datasetsChart = new ChartSeries();
-        datasetsChart.setLabel(BundleUtil.getStringFromBundle("metrics.datasets"));
+        String xLabel = BundleUtil.getStringFromBundle("metrics.month");
+        String yLabel = BundleUtil.getStringFromBundle("metrics.chart.legend.label." + chartType);
+        String title = BundleUtil.getStringFromBundle("metrics.chart.title." + chartType);
 
-        datasets.forEach(datasetStats ->
-                                 datasetsChart.set(datasetStats.getYear(), datasetStats.getCount()));
-
-        model.addSeries(datasetsChart);
-
-        return model;
+        return createBarModel(monthlyChartStats, title, xLabel, yLabel, chartMode);
     }
 
-    BarChartModel initMonthlyBarModel(List<DatasetsMetrics> datasets) {
+    BarChartModel initBarModel(List<ChartMetrics> metrics, String columnLabel, String chartMode) {
         BarChartModel model = new BarChartModel();
-        ChartSeries datasetsChart = new ChartSeries();
-        datasetsChart.setLabel("Datasets");
-
-        datasets.forEach(datasetStats ->
-                                 datasetsChart.set(BundleUtil.getStringFromBundle("metrics.month-" + datasetStats.getMonth()),
-                                                   datasetStats.getCount()));
-
-        model.addSeries(datasetsChart);
+        ChartSeries chartSeries = new ChartSeries();
+        chartSeries.setLabel(columnLabel);
+
+        if (chartMode.equals("YEAR")) {
+            metrics.forEach(metric ->
+                    chartSeries.set(metric.getYear(), metric.getCount()));
+        } else if (chartMode.equals("MONTH")) {
+            metrics.forEach(metric ->
+                    chartSeries.set(BundleUtil.getStringFromBundle("metrics.month-" + metric.getMonth()),
+                            metric.getCount()));
+        }
```

Comment:
I would refactor this to get rid of string comparing in if else. It is easy to make a mistake when comparing strings like that.
I would make two methods out of this:
`ChartSeries createYearlySeries(List<ChartMetrics> yearlyMetrics, String columnLabel)` and
`ChartSeries createMonthlySeries(List<ChartMetrics> monthlyMetrics, String columnLabel)`
and extract usage method from createBarModel to createMonthlyChart and createYearlyChart like
this:
`
BarChartModel model = createBarModel(...);
model.addSeries(createYearlySeries(...));
`
As an alternative i would use enum instead of strings

Reply:
I've kinda made this whole thing to avoid having parameter of type `BarChartModel` in method `createBarChart(...)` but I guess it's cleaner the way you propose it.